### ISSUE TITLE
Code Optimization and Modularization

### ISSUE THREAD

**Issue by kapoorp99** [sanchit0496/metricboard#1]
@sanchit0496 this is a really nice and straightforward middleware, i have sugesstions to make its codebase more modular by making different modules for each task, for example at first we can separate the helper functions from the main code and later on create separate files for each task and export the required functions.

**Comment by sanchit0496** [comment]
@kapoorp99  Seems like a good suggestion for V2, can you provide a list of helper function that we can create, to increase the modularity of the middleware

**Comment by kapoorp99** [comment]
@sanchit0496 yes sure, can you please add more steps to let me know how to setup the dev and test environment for local development
